src/Cardano/Shell/NodeIPC/Example.hs

```diff
@@ -59,42 +59,52 @@ exampleWithFD = do
 
     (clientReadHandle, clientWriteHandle) <- getReadWriteHandles
 
-    (_, responses) <-
-        ipcServer clientWriteHandle Ping
+    (_, responses) <- do
+        (serverReadHandle, serverWriteHandle) <- getReadWriteHandles
+        -- Send message to server
+        sendMessage serverWriteHandle Ping
+        startIPC SingleMessage serverReadHandle clientWriteHandle nodePort
         `concurrently`
         receieveMessages clientReadHandle
 
     return responses
 
--- | Example of an IPC using process
+-- | Example of an IPC using nc process
 exampleWithProcess :: IO (MsgOut, MsgOut)
-exampleWithProcess = do
-    (clientReadHandle, clientWriteHandle) <- getReadWriteHandles
-
-    -- Create a child process that acts as an server
+exampleWithProcess = bracket acquire restore action
+  where
+    acquire :: IO (ReadHandle, Handle)
+    acquire = do
+        (rFd, wFd) <- createPipeFd
+        setEnv "FD_WRITE_HANDLE" (show wFd)
+        readHandle  <- ReadHandle <$> fdToHandle rFd
+        -- Since closeFd only exists in 'unix' library,
+        -- the only way to close this Fd is to convert it into Handle and apply
+        -- hClose to it
+        writeHandle <- fdToHandle wFd
+        return (readHandle, writeHandle)
+
+    restore :: (ReadHandle, Handle) -> IO ()
+    restore ((ReadHandle rHandle), wHandle) = do
+        hClose rHandle
+        hClose wHandle
+        unsetEnv "FD_WRITE_HANDLE"
+
+    action :: (ReadHandle, Handle) -> IO (MsgOut, MsgOut)
+    action (readHandle, _) = do
+        withCreateProcess (proc "stack" ["exec", "node-ipc", "haskell"]) $
+        -- Tried to use stdin, but it is very incosistent.
+        -- the executable throws exception:
+        -- node-ipc: Unable to read with given handle: {handle: <stdout>}
+        -- Even though I'm providing stdin
+            \_ _ _ _ -> receieveMessages readHandle
```

Comment:
If you want to send the message once, you can use `void`, otherwise, you can use `forever`.